In [1]:
import pandas as pd 
import numpy as np 
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
tmdb_df = pd.read_csv('../../data/TMDB_tv_dataset_v3.csv')

In [3]:
tmdb_df

id                  name  number_of_seasons  number_of_episodes  \
0         1399       Game of Thrones                  8                  73   
1        71446           Money Heist                  3                  41   
2        66732       Stranger Things                  4                  34   
3         1402      The Walking Dead                 11                 177   
4        63174               Lucifer                  6                  93   
...        ...                   ...                ...                 ...   
158561  235768              天罚-二战全纪实                  1                 100   
158562  235767                 我们这一摊                  1                  13   
158563  235766               山里来了熊孩子                  1                  13   
158564  235765    开心超人联盟之勇敢之城1——棋兽传奇                  1                   6   
158565   38212  Anhoniyon Ka Andhera                  0                   0   

       original_language  vote_count  vote_average  \
0                     en       21857         8.442   
1                     es       17836         8.257   
2                     en       16161         8.624   
3                     en       15432         8.121   
4                     en       13870         8.486   
...                  ...         ...           ...   
158561                zh           0         0.000   
158562                zh           0         0.000   
158563                zh           0         0.000   
158564                zh           0         0.000   
158565                en           0         0.000   

                                                 overview  adult  \
0       Seven noble families fight for control of the ...  False   
1       To carry out the biggest heist in history, a m...  False   
2       When a young boy vanishes, a small town uncove...  False   
3       Sheriff's deputy Rick Grimes awakens from a co...  False   
4       Bored and unhappy as the Lord of Hell, Lucifer...  False   
...                                                   ...    ...   
158561                                                NaN  False   
158562                                                NaN  False   
158563                                                NaN  False   
158564                                                NaN  False   
158565  Anhoniyon Ka Andhera is a weekly Indian televi...  False   

                           backdrop_path  ...  \
0        /2OMB0ynKlyIenMJWI2Dy9IWT4c.jpg  ...   
1       /gFZriCkpJYsApPZEF3jhxL4yLzG.jpg  ...   
2       /2MaumbgBlW1NoPo3ZJO38A6v7OS.jpg  ...   
3       /x4salpjB11umlUOltfNvSSrjSXm.jpg  ...   
4       /aDBRtunw49UF4XmqfyNuD9nlYIu.jpg  ...   
...                                  ...  ...   
158561                               NaN  ...   
158562                               NaN  ...   
158563  /qAdXlecuPyg2jb2GbRXzHEUcpmw.jpg  ...   
158564  /ksuAmIZRkBMshJtkHtMDwyYLmDh.jpg  ...   
158565                               NaN  ...   

                                 tagline  \
0                       Winter Is Coming   
1                   The perfect robbery.   
2          Every ending has a beginning.   
3       Fight the dead. Fear the living.   
4                   It's good to be bad.   
...                                  ...   
158561                               NaN   
158562                               NaN   
158563                               NaN   
158564                               NaN   
158565                               NaN   

                                             genres  \
0       Sci-Fi & Fantasy, Drama, Action & Adventure   
1                                      Crime, Drama   
2                  Drama, Sci-Fi & Fantasy, Mystery   
3       Action & Adventure, Drama, Sci-Fi & Fantasy   
4                           Crime, Sci-Fi & Fantasy   
...                                             ...   
158561                  Documentary, War & Politics   
158562                                 

In [4]:
pd.set_option('display.max_columns', None)

In [5]:
tmdb_df

id                  name  number_of_seasons  number_of_episodes  \
0         1399       Game of Thrones                  8                  73   
1        71446           Money Heist                  3                  41   
2        66732       Stranger Things                  4                  34   
3         1402      The Walking Dead                 11                 177   
4        63174               Lucifer                  6                  93   
...        ...                   ...                ...                 ...   
158561  235768              天罚-二战全纪实                  1                 100   
158562  235767                 我们这一摊                  1                  13   
158563  235766               山里来了熊孩子                  1                  13   
158564  235765    开心超人联盟之勇敢之城1——棋兽传奇                  1                   6   
158565   38212  Anhoniyon Ka Andhera                  0                   0   

       original_language  vote_count  vote_average  \
0                     en       21857         8.442   
1                     es       17836         8.257   
2                     en       16161         8.624   
3                     en       15432         8.121   
4                     en       13870         8.486   
...                  ...         ...           ...   
158561                zh           0         0.000   
158562                zh           0         0.000   
158563                zh           0         0.000   
158564                zh           0         0.000   
158565                en           0         0.000   

                                                 overview  adult  \
0       Seven noble families fight for control of the ...  False   
1       To carry out the biggest heist in history, a m...  False   
2       When a young boy vanishes, a small town uncove...  False   
3       Sheriff's deputy Rick Grimes awakens from a co...  False   
4       Bored and unhappy as the Lord of Hell, Lucifer...  False   
...                                                   ...    ...   
158561                                                NaN  False   
158562                                                NaN  False   
158563                                                NaN  False   
158564                                                NaN  False   
158565  Anhoniyon Ka Andhera is a weekly Indian televi...  False   

                           backdrop_path first_air_date last_air_date  \
0        /2OMB0ynKlyIenMJWI2Dy9IWT4c.jpg     2011-04-17    2019-05-19   
1       /gFZriCkpJYsApPZEF3jhxL4yLzG.jpg     2017-05-02    2021-12-03   
2       /2MaumbgBlW1NoPo3ZJO38A6v7OS.jpg     2016-07-15    2022-07-01   
3       /x4salpjB11umlUOltfNvSSrjSXm.jpg     2010-10-31    2022-11-20   
4       /aDBRtunw49UF4XmqfyNuD9nlYIu.jpg     2016-01-25    2021-09-10   
...                                  ...            ...           ...   
158561                               NaN     2015-01-01    2023-09-29   
158562                               NaN     2023-09-02    2023-09-02   
158563  /qAdXlecuPyg2jb2GbRXzHEUcpmw.jpg     2023-09-16    2023-09-16   
158564  /ksuAmIZRkBMshJtkHtMDwyYLmDh.jpg     2023-09-26    2023-09-27   
158565                               NaN            NaN           NaN   

                                                 homepage  in_production  \
0                      http://www.hbo.com/game-of-thrones          False   
1                  https://www.netflix.com/title/80192098          False   
2                  https://www.netflix.com/title/80057281           True   
3       http://www.amc.com/shows/the-walking-dead--100...          False   
4                  https://www.netflix.com/title/80057918          False   
...                                                   ...            ...   
158561                                                NaN           True   
158562                                                NaN           True   
158563                              

In [6]:
tmdb_df=tmdb_df.drop(columns=['backdrop_path', 'homepage', 'poster_path'])
tmdb_df

id                  name  number_of_seasons  number_of_episodes  \
0         1399       Game of Thrones                  8                  73   
1        71446           Money Heist                  3                  41   
2        66732       Stranger Things                  4                  34   
3         1402      The Walking Dead                 11                 177   
4        63174               Lucifer                  6                  93   
...        ...                   ...                ...                 ...   
158561  235768              天罚-二战全纪实                  1                 100   
158562  235767                 我们这一摊                  1                  13   
158563  235766               山里来了熊孩子                  1                  13   
158564  235765    开心超人联盟之勇敢之城1——棋兽传奇                  1                   6   
158565   38212  Anhoniyon Ka Andhera                  0                   0   

       original_language  vote_count  vote_average  \
0                     en       21857         8.442   
1                     es       17836         8.257   
2                     en       16161         8.624   
3                     en       15432         8.121   
4                     en       13870         8.486   
...                  ...         ...           ...   
158561                zh           0         0.000   
158562                zh           0         0.000   
158563                zh           0         0.000   
158564                zh           0         0.000   
158565                en           0         0.000   

                                                 overview  adult  \
0       Seven noble families fight for control of the ...  False   
1       To carry out the biggest heist in history, a m...  False   
2       When a young boy vanishes, a small town uncove...  False   
3       Sheriff's deputy Rick Grimes awakens from a co...  False   
4       Bored and unhappy as the Lord of Hell, Lucifer...  False   
...                                                   ...    ...   
158561                                                NaN  False   
158562                                                NaN  False   
158563                                                NaN  False   
158564                                                NaN  False   
158565  Anhoniyon Ka Andhera is a weekly Indian televi...  False   

       first_air_date last_air_date  in_production         original_name  \
0          2011-04-17    2019-05-19          False       Game of Thrones   
1          2017-05-02    2021-12-03          False      La Casa de Papel   
2          2016-07-15    2022-07-01           True       Stranger Things   
3          2010-10-31    2022-11-20          False      The Walking Dead   
4          2016-01-25    2021-09-10          False               Lucifer   
...               ...           ...            ...                   ...   
158561     2015-01-01    2023-09-29           True              天罚-二战全纪实   
158562     2023-09-02    2023-09-02           True                 我们这一摊   
158563     2023-09-16    2023-09-16           True               山里来了熊孩子   
158564     2023-09-26    2023-09-27           True    开心超人联盟之勇敢之城1——棋兽传奇   
158565            NaN           NaN          False  Anhoniyon Ka Andhera   

        popularity         type            status  \
0         1083.917     Scripted             Ended   
1           96.354     Scripted             Ended   
2          185.711     Scripted  Returning Series   
3          489.746     Scripted             Ended   
4          416.668     Scripted             Ended   
...            ...          ...               ...   
158561      11.114  Documentary  Returning Series   
158562       0.762     Scripted  Returning Series   
158563       1.375     Scripted  Returning Series   
158564       3.547     Scripted  Returning Series   
158565       0.600     Scripted             Ended   

                                 tagline  \
0  

In [7]:
tmdb_voteavg=tmdb_df.sort_values(by=['vote_average'], ascending=False)
tmdb_voteavg.head(5000)

id                              name  number_of_seasons  \
47608  103628                   Адмиралы района                  2   
51253  128917    Белка и Стрелка: Тайны космоса                  1   
37581  212632                          Mørklagt                  1   
37582   81486  Bug Juice: My Adventures at Camp                  1   
23153  222838                         Колотушка                  1   
...       ...                               ...                ...   
42761  100870                            Tunnel                  1   
42758   58290         The Magic of the Big Blue                  1   
42757   58263                                O'                  8   
49361     757                       Finger Tips                  0   
42750   21765                      The Kagestar                  1   

       number_of_episodes original_language  vote_count  vote_average  \
47608                  36                ru           1          10.0   
51253                  20                ru           1          10.0   
37581                   1                da           1          10.0   
37582                  16                en           1          10.0   
23153                  10                ru           3          10.0   
...                   ...               ...         ...           ...   
42761                  16                id           1          10.0   
42758                   7                en           1          10.0   
42757                 174                fr           1          10.0   
49361                   0                en           1          10.0   
42750                  34                ja           1          10.0   

                                                overview  adult  \
47608                                                NaN  False   
51253                                                NaN  False   
37581                                                NaN  False   
37582  Return to Camp Waziyatah for a contemporary ne...  False   
23153                                                NaN  False   
...                                                  ...    ...   
42761  The life of a reliable detective named Tigor c...  False   
42758                                                NaN  False   
42757                                                NaN  False   
49361  Finger Tips is a television programme produced...  False   
42750  While rescuing his wealthy boss's daughter fro...  False   

      first_air_date last_air_date  in_production  \
47608     2020-05-18    2022-12-12           True   
51253     2019-01-01    2019-05-14           True   
37581     2022-10-18    2022-10-18           True   
37582     2018-07-16    2018-08-09          False   
23153     2023-03-23    2023-04-20           True   
...              ...           ...            ...   
42761     2019-12-23    2020-03-30          False   
42758     2013-02-15    2013-03-29           True   
42757     2012-01-24    2019-03-26          False   
49361            NaN           NaN          False   
42750     1976-04-05    1976-11-29          False   

                          original_name  popularity         type  \
47608                   Адмиралы района       1.088     Scripted   
51253    Белка и Стрелка: Тайны космоса       0.601     Scripted   
37581                          Mørklagt       0.600     Scripted   
37582  Bug Juice: My Adventures at Camp       0.999  Documentary   
23153                         Колотушка       0.811     Scripted   
...                                 ...         ...          ...   
42761                            Tunnel       3.458     Scripted   
42758         The Magic of the Big Blue       2.842  Documentary   
42757                                O'      27.997     Scripted   
49361                       Finger Tips       0.600     Scripted   
42750                           ザ・カゲスター       2.295     Scripted   

                 status tagline   

In [8]:
tmdb_topvoteavg = tmdb_df[tmdb_df['vote_average']==10.0]
tmdb_topvoteavg

id                            name  number_of_seasons  \
5722    105696  Scooby-Doo!: Behind the Scenes                  1   
8111    210805            Tyler Perry's Zatima                  3   
10811    84274                        Delirium                  1   
11063    93294               Dog's Most Wanted                  1   
12198    71165                  Dateline on ID                 11   
...        ...                             ...                ...   
133986   62546               When We Go to War                  1   
141129   76797                  Mai dire isola                  1   
144271  201403                   My Money & Me                  1   
147045  210051                    Hole in Love                  1   
152497  207194                Court Night Live                  1   

        number_of_episodes original_language  vote_count  vote_average  \
5722                     8                en          34          10.0   
8111                    26                en          19          10.0   
10811                    0                en          11          10.0   
11063                   10                en          11          10.0   
12198                   67                en           9          10.0   
...                    ...               ...         ...           ...   
133986                   6                en           0          10.0   
141129                   9                it           0          10.0   
144271                   6                en           0          10.0   
147045                  14                ko           0          10.0   
152497                  12                en           0          10.0   

                                                 overview  adult  \
5722    A series of eight shorts that aired on Cartoon...  False   
8111    Zac and Fatima take a huge step to strengthen ...  False   
10811   Set in a world where love is deemed illegal an...  False   
11063                                                 NaN  False   
12198               Criminal investigations are detailed.  False   
...                                                   ...    ...   
133986  A historical saga, it tells the story of six y...  False   
141129                    Reality commented by Gialappa's  False   
144271                                                NaN  False   
147045  'Golf' and 'love' have something in common and...  False   
152497  Court Night Live brings live trials to the peo...  False   

       first_air_date last_air_date  in_production  \
5722       1998-01-01    1998-02-19          False   
8111       2022-09-21    2023-09-07           True   
10811      2014-06-20    2014-06-20           True   
11063      2019-09-04    2019-11-06           True   
12198             NaN    2017-08-29           True   
...               ...           ...            ...   
133986     2015-04-26    2015-05-24          False   
141129     2018-02-15    2018-04-12          False   
144271     2016-01-01    2016-01-01           True   
147045     2022-01-31    2022-04-25          False   
152497     2022-08-10    2022-10-26           True   

                         original_name  popularity         type  \
5722    Scooby-Doo!: Behind the Scenes       5.165   Miniseries   
8111              Tyler Perry's Zatima       9.992     Scripted   
10811                         Delirium       6.278     Scripted   
11063                Dog's Most Wanted       0.947      Reality   
12198                   Dateline on ID       1.524     Scripted   
...                                ...         ...          ...   
133986               When We Go to War       0.781   Miniseries   
141129                  Mai dire isola       0.633     Scripted   
144271                   My Money & Me       0.600  Documentary   
147045                            홀인러브       0.692      Reality   
152497                Court Night Live       1.673      Reality   

                  statu

## 5221 have a average voter rating of 10.0

In [9]:
tmdb_noseasons = tmdb_df[tmdb_df['number_of_seasons']==0]
tmdb_noseasons

id                   name  number_of_seasons  number_of_episodes  \
1972    20727             La Mentira                  0                   0   
2451    33148  En Los Tacones De Eva                  0                   0   
3338     2807            La Traición                  0                   0   
3945    12889             Tú o nadie                  0                   0   
4109     8067            Live XXX TV                  0                   0   
...       ...                    ...                ...                 ...   
158549  38214            Homemade TV                  0                   0   
158550  38213     Sunday Pops Series                  0                   0   
158552  38209     All Over the Place                  0                   0   
158553  38208     All Star Glam Exam                  0                   0   
158565  38212   Anhoniyon Ka Andhera                  0                   0   

       original_language  vote_count  vote_average  \
1972                  es         178         7.200   
2451                  en         131         7.600   
3338                  en          86         7.600   
3945                  en          66         6.909   
4109                  en          62         4.200   
...                  ...         ...           ...   
158549                en           0         0.000   
158550                en           0         0.000   
158552                en           0         0.000   
158553                en           0         0.000   
158565                en           0         0.000   

                                                 overview  adult  \
1972    La Mentira is a Mexican telenovela, produced b...  False   
2451                                                  NaN  False   
3338    La Traición is a Colombian-American telenovela...  False   
3945    Tú o nadie is a Mexican telenovela. It was pro...  False   
4109    Live XXX TV was an interactive televised sex l...  False   
...                                                   ...    ...   
158549  Homemade TV is a Canadian children's televisio...  False   
158550  Sunday Pops Series is a Canadian classical mus...  False   
158552  All Over the Place is a children's television ...  False   
158553  All Star Glam Exam is a Hong Kong game show pr...  False   
158565  Anhoniyon Ka Andhera is a weekly Indian televi...  False   

       first_air_date last_air_date  in_production          original_name  \
1972              NaN           NaN          False             La Mentira   
2451              NaN           NaN          False  En Los Tacones De Eva   
3338              NaN           NaN          False            La Traición   
3945              NaN           NaN          False             Tú o nadie   
4109              NaN           NaN          False            Live XXX TV   
...               ...           ...            ...                    ...   
158549            NaN           NaN          False            Homemade TV   
158550            NaN           NaN          False     Sunday Pops Series   
158552            NaN           NaN          False     All Over the Place   
158553            NaN           NaN          False     All Star Glam Exam   
158565            NaN           NaN          False   Anhoniyon Ka Andhera   

        popularity      type status tagline genres created_by languages  \
1972         4.037  Scripted  Ended     NaN  Drama        NaN        es   
2451         3.823  Scripted  Ended     NaN    NaN        NaN       NaN   
3338         4.514  Scripted  Ended     NaN  Drama        NaN        en   
3945         2.950  Scripted  Ended     NaN    NaN        NaN       NaN   
4109         2.647  Scripted  Ended     NaN    NaN        NaN       NaN   
...            ...       ...    ...     ...    ...        ...       ...   
158549       0.600  Scripted  Ended     NaN    NaN        NaN        en   
158550       0.600  Scripted  Ended     NaN    NaN        NaN        en

## 22378 have no seasons. Not sure how that's a thing.

In [10]:
tmdb_noepisodes = tmdb_df[tmdb_df['number_of_episodes']==0]
tmdb_noepisodes

id                                          name  \
1972     20727                                    La Mentira   
2088    113558                        Pandillas guerra y paz   
2451     33148                         En Los Tacones De Eva   
3338      2807                                   La Traición   
3945     12889                                    Tú o nadie   
...        ...                                           ...   
158550   38213                            Sunday Pops Series   
158552   38209                            All Over the Place   
158553   38208                            All Star Glam Exam   
158555  235758  Alya Sometimes Hides Her Feelings in Russian   
158565   38212                          Anhoniyon Ka Andhera   

        number_of_seasons  number_of_episodes original_language  vote_count  \
1972                    0                   0                es         178   
2088                    1                   0                es         166   
2451                    0                   0                en         131   
3338                    0                   0                en          86   
3945                    0                   0                en          66   
...                   ...                 ...               ...         ...   
158550                  0                   0                en           0   
158552                  0                   0                en           0   
158553                  0                   0                en           0   
158555                  1                   0                ja           0   
158565                  0                   0                en           0   

        vote_average                                           overview  \
1972           7.200  La Mentira is a Mexican telenovela, produced b...   
2088           8.300                                                NaN   
2451           7.600                                                NaN   
3338           7.600  La Traición is a Colombian-American telenovela...   
3945           6.909  Tú o nadie is a Mexican telenovela. It was pro...   
...              ...                                                ...   
158550         0.000  Sunday Pops Series is a Canadian classical mus...   
158552         0.000  All Over the Place is a children's television ...   
158553         0.000  All Star Glam Exam is a Hong Kong game show pr...   
158555         0.000  Masachika Kuse sits next to Arya, a girl of Ru...   
158565         0.000  Anhoniyon Ka Andhera is a weekly Indian televi...   

        adult first_air_date last_air_date  in_production  \
1972    False            NaN           NaN          False   
2088    False            NaN           NaN           True   
2451    False            NaN           NaN          False   
3338    False            NaN           NaN          False   
3945    False            NaN           NaN          False   
...       ...            ...           ...            ...   
158550  False            NaN           NaN          False   
158552  False            NaN           NaN          False   
158553  False            NaN           NaN          False   
158555  False            NaN           NaN           True   
158565  False            NaN           NaN          False   

                 original_name  popularity      type            status  \
1972                La Mentira       4.037  Scripted             Ended   
2088    Pandillas guerra y paz       6.463  Scripted  Returning Series   
2451     En Los Tacones De Eva       3.823  Scripted             Ended   
3338               La Traición       4.514  Scripted             Ended   
3945                Tú o nadie       2.950  Scripted             Ended   
...                        ...         ...       ...               ...   
158550      Sunday Pops Series       0.600  Scripted             Ended   
158552      All Over the Place       0.600  Scripted             End

## 23973 have no episodes.

In [11]:
tmdb_lang=tmdb_df.original_language.value_counts()
tmdb_lang

en    73638
ja    13368
zh    12469
ko     7384
fr     6926
      ...  
sw        1
st        1
se        1
mn        1
aa        1
Name: original_language, Length: 106, dtype: int64

## 106 languages, english in first with 73638, japanese in second with 13368, and zh apparently means chinese which is third with 12469

In [12]:
tmdb_df.vote_count.describe()


count    158566.000000
mean         14.144300
std         196.746395
min           0.000000
25%           0.000000
50%           0.000000
75%           1.000000
max       21857.000000
Name: vote_count, dtype: float64

In [14]:
tmdb_df.adult.value_counts()

False    156628
True       1938
Name: adult, dtype: int64

## 1938 are categorized as adult shows

In [15]:
tmdb_df.number_of_seasons.value_counts()

1      104943
0       22378
2       14015
3        5875
4        3194
        ...  
134         1
101         1
70          1
95          1
111         1
Name: number_of_seasons, Length: 82, dtype: int64

In [18]:
tmdb_df[tmdb_df['number_of_seasons']==240]

id           name  number_of_seasons  number_of_episodes  \
7088  6480  House Hunters                240                 804   

     original_language  vote_count  vote_average  \
7088                en          24           5.3   

                                               overview  adult first_air_date  \
7088  Hosted by Suzanne Whang, the show takes viewer...  False     1999-09-30   

     last_air_date  in_production  original_name  popularity     type  \
7088    2023-10-02           True  House Hunters       38.52  Reality   

                status tagline   genres created_by languages networks  \
7088  Returning Series     NaN  Reality        NaN        en     HGTV   

     origin_country spoken_languages  production_companies  \
7088             US          English  Pie Town Productions   

          production_countries  episode_run_time  
7088  United States of America                22

In [17]:
tmdb_df.number_of_seasons.describe()

count    158566.000000
mean          1.566761
std           3.005675
min           0.000000
25%           1.000000
50%           1.000000
75%           1.000000
max         240.000000
Name: number_of_seasons, dtype: float64

## House hunters has the most seasons with 240

In [19]:
tmdb_df.number_of_episodes.describe()

count    158566.000000
mean         25.388084
std         138.793401
min           0.000000
25%           1.000000
50%           6.000000
75%          20.000000
max       20839.000000
Name: number_of_episodes, dtype: float64

In [20]:
tmdb_df[tmdb_df['number_of_episodes']==20839]

id        name  number_of_seasons  number_of_episodes  \
2284  94722  Tagesschau                 72               20839   

     original_language  vote_count  vote_average  \
2284                de         145           7.5   

                                               overview  adult first_air_date  \
2284  German daily news program, the oldest still ex...  False     1952-12-26   

     last_air_date  in_production original_name  popularity  type  \
2284    2023-10-02           True    Tagesschau    3049.228  News   

                status tagline genres created_by languages networks  \
2284  Returning Series     NaN   News        NaN        de      ARD   

     origin_country spoken_languages production_companies  \
2284             DE          Deutsch                  NDR   

     production_countries  episode_run_time  
2284              Germany                15

## A german daily news show called Tagesschau has the highest number of episodes with 20839. It has been on air since 1952.

In [25]:
tmdb_df.first_air_date.sort_values()

32331     1917-01-09
118719    1921-01-01
55213     1936-11-03
40166     1938-07-12
45356     1938-08-01
             ...    
158550           NaN
158552           NaN
158553           NaN
158555           NaN
158565           NaN
Name: first_air_date, Length: 158566, dtype: object

In [29]:
tmdb_df[tmdb_df['first_air_date']=='1917-01-09']

id                                         name  number_of_seasons  \
32331  156597  B-Movies: Las obras maestras del cine cutre                  2   

       number_of_episodes original_language  vote_count  vote_average  \
32331                   1                es           2           8.0   

      overview  adult first_air_date last_air_date  in_production  \
32331      NaN  False     1917-01-09    1917-01-09           True   

                                     original_name  popularity      type  \
32331  B-Movies: Las obras maestras del cine cutre       0.613  Scripted   

                 status tagline       genres created_by languages  \
32331  Returning Series     NaN  Documentary        NaN       NaN   

            networks origin_country spoken_languages  \
32331  ARTE Creative             ES              NaN   

                   production_companies production_countries  episode_run_time  
32331  CNC, Nanarland, TSVP Productions                  NaN                 0

## B-Movies: Las obras maestras del cine cutre is the earliest show with the first air date of January 9, 1917

In [32]:
tmdb_df.last_air_date.sort_values(ascending=False)

81094     2023-10-03
45528     2023-10-03
113038    2023-10-03
92639     2023-10-03
118919    2023-10-03
             ...    
158550           NaN
158552           NaN
158553           NaN
158555           NaN
158565           NaN
Name: last_air_date, Length: 158566, dtype: object

In [33]:
tmdb_df[tmdb_df['last_air_date']=='2023-10-03']

id                                       name  number_of_seasons  \
2632    209265                             Land of Desire                  1   
3051    215803                              Batang Quiapo                  2   
4132    126929                         Welcome to Wrexham                  2   
4306    226411                            Revenge is mine                  1   
4412    229932      Pyaar Ka Pehla Adhyaya  - Shiv Shakti                  1   
...        ...                                        ...                ...   
158408  230518  Enquêtes  Paranormale de la rivière Grand                  1   
158474  235781               The Vintage French Farmhouse                  1   
158520  235777                                        Ира                  1   
158535  235730                                My Precious                  1   
158560  235769                                   Obituary                  1   

        number_of_episodes original_language  vote_count  vote_average  \
2632                   150                pt         120         6.600   
3051                   178                tl          99         7.030   
4132                    27                en          61         8.279   
4306                   110                es          57         7.100   
4412                    90                hi          55         7.864   
...                    ...               ...         ...           ...   
158408                  13                fr           0         0.000   
158474                  20                en           0         0.000   
158520                  10                ru           0         0.000   
158535                  40                zh           0         0.000   
158560                   6                en           0         0.000   

                                                 overview  adult  \
2632    When her husband is killed in a land grabbing ...  False   
3051    A young man rises to be one of the biggest out...  False   
4132    Documentary series tracking the dreams and wor...  False   
4306                                                  NaN  False   
4412    Shiv has a rewarding career as a doctor, but h...  False   
...                                                   ...    ...   
158408                                                NaN  False   
158474  Three antique dealers head to a French farmhou...  False   
158520                                                NaN  False   
158535  Tao You You is a veterinarian who dreamed of s...  False   
158560  An obituarist working at a small-town newspape...  False   

       first_air_date last_air_date  in_production  \
2632       2023-05-08    2023-10-03           True   
3051       2023-02-13    2023-10-03           True   
4132       2022-08-24    2023-10-03           True   
4306       2023-06-12    2023-10-03           True   
4412       2023-07-03    2023-10-03           True   
...               ...           ...            ...   
158408     2023-07-11    2023-10-03           True   
158474     2023-09-25    2023-10-03           True   
158520     2023-09-25    2023-10-03           True   
158535     2023-10-03    2023-10-03           True   
158560     2023-09-26    2023-10-03           True   

                                    original_name  popularity         type  \
2632                               Terra e Paixão    1937.067     Scripted   
3051                                   Quiapo Kid    3065.138     Scripted   
4132                           Welcome to Wrexham      67.743  Documentary   
4306                           Mía es la venganza    2207.068     Scripted   
4412        Pyaar Ka Pehla Adhyaya  - Shiv Shakti    1195.145     Scripted   
...                                           ...         ...          ...   
158408  Enquêtes  Paranormale de la rivière Grand      22.760     Scripted   
158474               The Vintage French Farmhouse      20.664      Reali

## There are 190 of them that are still airing as of October 3, 2023